services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/RecordHandler.java

```diff
@@ -842,13 +1003,26 @@ private void doGetExpectedDefaultRulesRequest(CommandMessage message) {
         IKafkaProducerService producerService = getKafkaProducer();
         String replyToTopic = context.getKafkaSwitchManagerTopic();
 
-        SwitchId switchId = ((GetExpectedDefaultRulesRequest) message.getData()).getSwitchId();
+        GetExpectedDefaultRulesRequest request = (GetExpectedDefaultRulesRequest) message.getData();
+        SwitchId switchId = request.getSwitchId();
+        boolean multiTable = request.isMultiTable();
+        List<Integer> islPorts = request.getIslPorts();
+        List<Integer> flowPorts = request.getFlowPorts();
 
         try {
             logger.debug("Loading expected default rules for switch {}", switchId);
-
+            DatapathId dpid = DatapathId.of(switchId.toLong());
             List<OFFlowMod> defaultRules =
-                    context.getSwitchManager().getExpectedDefaultFlows(DatapathId.of(switchId.toLong()));
+                    context.getSwitchManager().getExpectedDefaultFlows(dpid, multiTable);
+            if (multiTable) {
+                for (int port : islPorts) {
+                    List<OFFlowMod> islFlows = context.getSwitchManager().getExpectedIslFlowsForPort(dpid, port);
+                    defaultRules.addAll(islFlows);
+                }
+                for (int port : flowPorts) {
+                    defaultRules.add(context.getSwitchManager().buildIntermediateIngressRule(dpid, port));
+                }
+            }
             List<FlowEntry> flows = defaultRules.stream()
                     .map(OfFlowStatsMapper.INSTANCE::toFlowEntry)
                     .collect(Collectors.toList());
```

Comment:
Would be nice to verify that on a switch with a lot of rules this response fits in a kafka record.

Reply:
As long as it works for dump rules during switch validation it should be fine.